src/java.desktop/share/classes/javax/swing/JFileChooser.java

```diff
@@ -416,6 +416,21 @@ private void installShowFilesListener() {
         }
     }
 
+    @Override
+    public void setEnabled(boolean enabled) {
```

Comment:
Can you please check why a similar example using JFrame and button works fine, but the JFileChooser does not work?
```
        JFrame frame = new JFrame();
        JButton button = new JButton("1234567890");
        frame.add(button);
        frame.setVisible(true);
        frame.setEnabled(false);
        boolean enabled = button.isEnabled();
        System.out.println("enabled = " + enabled);
```

Reply:
JFileChooser added to a JFrame similar to the example also works, I guess it's because frame is made inaccessible. But when we try to disable JFileChooser alone, it's not working because we need to disable each sub-component in which JFileChooser is made up of. As per debugging I found out that peer component is showing null for JFileChooser.disable whereas for JFrame the disabling is taken care by peer component in Component class.

Reply:
Why we cannot make inaccessible JFileChooser same as JFrame? We do not need to mark disable allcomponents inside the frame(I guess jdialog and windows as well). Both are top level components, and probably should work in the same way.

Reply:
I guess for frame, the disable/enable functionality is handled by the peer (Windows it is `WFramePeer`), whereas for the swing components it should be the components itself like it is for all other components (Like in AbstractButton, JComboBox, JList).

Reply:
The JFileChooser is also kind of top level component and it has the peer. Current patch does not work as intended. On macOS in Nimbus/Metal L&F the list of files looks disabled but scrolling works. On Aqua L&F the list of files does not look like disabled, accept scroll and double clicks, but does not accept single clicks.

Reply:
I missed checking the scrolling part in macos, will check and get back on it.

Reply:
@mrserb, you are right on double and scrolling still working after the fix. I'm working on debugging the root cause, which I think something to do with filepane class. Parallel to that working on your comment previous comments too.